.github/workflows/ci.yaml

```diff
@@ -0,0 +1,53 @@
+name: CI
+
+on:
+  push:
+    branches: # only for pushes on main
+      - main
+  pull_request: # for all PRs regardless of its base branch
+
+jobs:
+  build:
+    # TODO: need to change self-hosted machine and GO env
+    name: Build & Test
+    runs-on: ubuntu-latest
```

Comment:
Good for now. I'll make it work on the self-hosted Github Actions runner (on Azure confidential computing).